src/monitor/monitor.c

```diff
@@ -231,10 +231,10 @@ static int client_registration(struct sbus_request *dbus_req, void *data)
     DBusError dbus_error;
     dbus_uint16_t svc_ver;
     dbus_uint16_t svc_type;
+    dbus_uint16_t explicitly_configured = true;
     char *svc_name;
     dbus_bool_t dbret;
     int ret;
-    bool explicitly_configured = true;
 
```

Comment:
So, I've done a quick test here and here is what the changes look like:
```
    diff --git a/src/monitor/monitor.c b/src/monitor/monitor.c
    index bf5d01f..381c46c 100644
    --- a/src/monitor/monitor.c
    +++ b/src/monitor/monitor.c
    @@ -231,7 +231,7 @@ static int client_registration(struct sbus_request *dbus_req, void *data)
         DBusError dbus_error;
         dbus_uint16_t svc_ver;
         dbus_uint16_t svc_type;
    -    dbus_uint16_t explicitly_configured = true;
    +    dbus_bool_t explicitly_configured = true;
         char *svc_name;
         dbus_bool_t dbret;
         int ret;
    @@ -331,7 +331,7 @@ static int client_registration(struct sbus_request *dbus_req, void *data)
         /* reply that all is ok */
         sbus_request_return_and_finish(dbus_req,
                                        DBUS_TYPE_UINT16, &version,
    -                                   DBUS_TYPE_UINT16, &explicitly_configured,
    +                                   DBUS_TYPE_BOOLEAN, &explicitly_configured,
                                        DBUS_TYPE_INVALID);
     
     done:
    diff --git a/src/monitor/monitor_sbus.c b/src/monitor/monitor_sbus.c
    index 3b09685..784a541 100644
    --- a/src/monitor/monitor_sbus.c
    +++ b/src/monitor/monitor_sbus.c
    @@ -51,7 +51,7 @@ static void id_callback(DBusPendingCall *pending, void *ptr)
         DBusError dbus_error;
         dbus_bool_t ret;
         dbus_uint16_t mon_ver;
    -    dbus_uint16_t explicitly_configured;
    +    dbus_bool_t explicitly_configured;
         int type;
     
         dbus_error_init(&dbus_error);
    @@ -76,7 +76,7 @@ static void id_callback(DBusPendingCall *pending, void *ptr)
     
             ret = dbus_message_get_args(reply, &dbus_error,
                                         DBUS_TYPE_UINT16, &mon_ver,
    -                                    DBUS_TYPE_UINT16, &explicitly_configured,
    +                                    DBUS_TYPE_BOOLEAN, &explicitly_configured,
                                         DBUS_TYPE_INVALID);
             if (!ret) {
                 DEBUG(SSSDBG_CRIT_FAILURE, "Failed to parse message\n");
```     

With these changes I'm always receiving that the service was explicitly configured, doesn't matter whether it was socket-activated or not. So, that was the reason I ended up using the UINT16 type.